source/frontend/properties/gc/TevSolver.cpp

```diff
@@ -165,6 +165,59 @@ static bool optimizeNodeBinary(BinExpr& e) {
     return true;
   }
 
+  // Axiom of identity
+  // 0 == 0 -> true
+  // 0 == 1 -> false
+  // 1 == 0 -> false
+  // 1 == 1 -> true
+  if (e.op == ExprOp::Eq) {
+    if ((left_zero && right_zero) || (left_one && right_one)) {
+      to_unary(One);
+      return true;
+    }
+    if ((left_one && right_zero) || (left_zero && right_one)) {
+      to_unary(Zero);
+      return true;
+    }
+  }
+
+  // 0 > 0 -> false
+  // 0 > 1 -> false
+  // 1 > 0 -> true
+  // 1 > 1 -> false
+  if (e.op == ExprOp::Gt) {
+    if ((left_zero && right_zero) || (left_zero && right_one) ||
+        (left_one && right_one)) {
+      to_unary(One);
+      return true;
+    }
+    if ((left_one && right_zero)) {
+      to_unary(Zero);
+      return true;
+    }
+  }
+
+  if (e.op == ExprOp::TernOp) {
+
+    // Bit of a hack: done to skip displaying a step
+    // A ? 0 : 0 -> 0
+    const auto rl_un = reinterpret_cast<UnaryExpr*>(right_bin->left);
```

Comment:
Style: don't use `const auto` to refer to a pointer/reference type; be more explicit with `const auto*` or `const auto&`.